feat: added support for custom i18n messages

rmathis opened the pull request:
Added support for passing in a json string for customizing the i18n messages. Also separated the i18n logic into a mixin for all components to use.

jgribonvald commented:
@rmathis it's a really nice improvment :angel: 
But you forgot the hamburger-menu, could you activate it from the App.vue and add the missing part ? And the "top notch" would be to have the documentation updated.
 
Thanks a lot

rmathis commented:
> @rmathis it's a really nice improvment 👼
> You forgot the hamburger-menu, could you activate it from the App.vue and add the missing part ?
> And the "top notch" thing would be to have the documentation updated.
> 
> Thanks a lot
> 
> Note: for the hamburger-menu testing you can also add the property show-favorites-in-slider to switch between slider and content-grid.

Updated docs and added the mixin to the hamburger menu
